### ISSUE TITLE
Thousand Land | Not Rendering Correctly

### ISSUE THREAD

**Issue by Solniz** [xemu-project/xemu#1026]
### Title

https://xemu.app/titles/46530003/#Thousand-Land

### Bug Description

Once you reach in game, everything is rendering in white, you can hardly tell what is going on

![xemu-2022-06-04-18-12-44](https://user-images.githubusercontent.com/44263219/172020154-7f8bb97d-9dbd-480a-8713-849fc680d94a.png)
![xemu-2022-06-04-18-13-05](https://user-images.githubusercontent.com/44263219/172020155-99c02af6-7075-4650-8cd3-821ccdb22d6d.png)

### Expected Behavior

The ingame graphics should be rendering correctly like this:

![thousandland4](https://user-images.githubusercontent.com/44263219/172020210-01e7899f-1d33-4b28-b633-ba0ee580bb23.png)

### xemu Version

0.7.28

### System Information

Windows 11
Intel i7-10875H
RTX 2080 8GB
512.95

**Comment by Triticum0** [comment]
I used debug build to get an output that might be useful
[log.txt](https://github.com/mborgerson/xemu/files/8991955/log.txt)

**Comment by abaire** [comment]
@Triticum0 you'll need to enable the nv2a tracing via the monitor rather than the debug define (the tracing now contains all of the interesting information).

You can do `trace-event nv2a_pgraph_* on` in the monitor window to turn it on (replacing `on` with `off` will turn it off afterwards)

**Comment by Triticum0** [comment]
@abaire Here you go
[xemu.log](https://github.com/mborgerson/xemu/files/8993226/xemu.log)
Also Tank performace some much have to kill the program is it meant to be that slow had better performace on my Pentium

**Comment by Triticum0** [comment]
Also when using Debug from the code actually useful? Should I never enable it and use trace-event from now on

**Comment by abaire** [comment]
> @abaire Here you go [xemu.log](https://github.com/mborgerson/xemu/files/8993226/xemu.log) Also Tank performace some much have to kill the program is it meant to be that slow had better performace on my Pentium
Thanks, will take a look.

It's expected that it slows things down, though in my experience it usually isn't enough to make things that unresponsive. You may be have better performance if you pipe the output directly to a file rather than to a console, but it probably also depends on the specifics of your machine and OS.


> Also when using Debug from the code actually useful? Should I never enable it and use trace-event from now on
These days it's only useful in very specific cases. I would suggest you just use the `trace-event` system unless a contributor specifically asks you to enable the other debugging options.

**Comment by abaire** [comment]
I don't see anything that jumps out at me in the log file. The unhandled commands look like the usual set, I do see some uses of stride:0 in positions and textures so this could be similar to #686 (maybe there are more nuances to the behavior than expected?).

**Comment by Triticum0** [comment]
Yeah I think it's a bug, my hardware better spec than yours I'm getting sub 1 fps on Windows something happen that shouldn't

**Comment by IsaacStitch** [comment]
Any fixes ?

**Comment by abaire** [comment]
If there were any fixes they'd be in a PR linked to this bug. I don't think any of contributors have this game, so likely nobody is looking at it.

**Comment by Kravataf** [comment]
bruh

**Comment by Enkak** [comment]
Hopefully this extra info can help somewhat. 

Since this game is from the same company, decided to do a little test using an older version of Xemu that included some PRs (https://github.com/xemu-project/xemu/issues/365) that can fix part of the Otogi graphical issues. That makes the ingame graphics appear in Thousand Land too instead of a white screen as described in the bug and still present on latest Xemu release. 

Not perfect but maybe some of those PRs not included yet might make it playable. 

![test1](https://github.com/xemu-project/xemu/assets/11450342/2549aee0-912d-4b3a-aaa7-5d6f8d98966d)

**Comment by Triticum0** [comment]
Thanks, don't know why I never thought to test it with that PR, probably because it never got merged. @abaire seems like the fix was simple.

Yeah. @Enkak thanks for testing it not sure if I ever get around to retesting it on the masters+Otogi fix as I stopped testing Xemu as the project has stopped updating to a point where it is not worth the effort testing the games. Solniz should add to the PR and will link to the Otogi Issue. Also forgot how to make a custom build from Git so can't help you there either.

**Comment by Kravataf** [comment]
> Hopefully this extra info can help somewhat.
> 
> Since this game is from the same company, decided to do a little test using an older version of Xemu that included some PRs (#365) that can fix part of the Otogi graphical issues. That makes the ingame graphics appear in Thousand Land too instead of a white screen as described in the bug and still present on latest Xemu release.
> 
> Not perfect but maybe some of those PRs not included yet might make it playable.
> 
> ![test1](https://private-user-images.githubusercontent.com/11450342/345402174-2549aee0-912d-4b3a-aaa7-5d6f8d98966d.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExNTExNjksIm5iZiI6MTcyMTE1MDg2OSwicGF0aCI6Ii8xMTQ1MDM0Mi8zNDU0MDIxNzQtMjU0OWFlZTAtOTEyZC00YjNhLWFhYTctNWQ2ZjhkOTg5NjZkLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE2VDE3Mjc0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRhMDcyMTI3MTkyOTAzOWFmY2MzYmExOGIxYmU0MmU3YTEyNGY1NWQzZmNlNWRhNzk1NmYzZThhZmQ1ZGZiOGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.FIZwpVqBWEFhDMQtcJLn5LIrtRJgwEQgRwU-GWPMMWs)

@enkak could you send the build of xemu please? i tried compiling but it just didnt work...

**Comment by Enkak** [comment]
> @Enkak could you send the build of xemu please? i tried compiling but it just didnt work...

Not sure if that is against the rules or dev wishes. Anyway at the moment I don't have the computer with me the next two weeks so can only do it after.

**Comment by Kravataf** [comment]
> > @Enkak could you send the build of xemu please? i tried compiling but it just didnt work...
> 
> Not sure if that is against the rules or dev wishes. Anyway at the moment I don't have the computer with me the next two weeks so can only do it after.

okay thank you :) until then im just going to keep trying to compile it without errors or crashes, until youll have your computer with you

edit: im stupid what i edited in xemu was completely wrong so when i tried to compile it thats why it always crashed :|

**Comment by Kravataf** [comment]
> > @Enkak could you send the build of xemu please? i tried compiling but it just didnt work...
> 
> Not sure if that is against the rules or dev wishes. Anyway at the moment I don't have the computer with me the next two weeks so can only do it after.

do you have your computer with you yet?

**Comment by Enkak** [comment]
> > > @Enkak could you send the build of xemu please? i tried compiling but it just didnt work...
> > 
> > 
> > Not sure if that is against the rules or dev wishes. Anyway at the moment I don't have the computer with me the next two weeks so can only do it after.
> 
> do you have your computer with you yet?

Hi. Please find attached the older version of Xemu and see how it goes.
[xemu.zip](https://github.com/user-attachments/files/16423657/xemu.zip)

**Comment by Kravataf** [comment]
> > > > @Enkak could you send the build of xemu please? i tried compiling but it just didnt work...
> > > 
> > > 
> > > Not sure if that is against the rules or dev wishes. Anyway at the moment I don't have the computer with me the next two weeks so can only do it after.
> > 
> > 
> > do you have your computer with you yet?
> 
> Hi. Please find attached the older version of Xemu and see how it goes. [xemu.zip](https://github.com/user-attachments/files/16423657/xemu.zip)

it works!! thank you so much :DD

**Comment by Kravataf** [comment]
also setting the rendering scale to 4 makes the lines appear a lot less!!

**Comment by IsaacStitch** [comment]
> > > > > @Enkak could you send the build of xemu please? i tried compiling but it just didnt work...
> > > > 
> > > > 
> > > > Not sure if that is against the rules or dev wishes. Anyway at the moment I don't have the computer with me the next two weeks so can only do it after.
> > > 
> > > 
> > > do you have your computer with you yet?
> > 
> > 
> > Hi. Please find attached the older version of Xemu and see how it goes. [xemu.zip](https://github.com/user-attachments/files/16423657/xemu.zip)
> 
> it works!! thank you so much :DD

So, no more graphical glitches ?
Can you post a screenshot ?

Thanks @Enkak
